cmd/subcmds/optimizeSG.go

```diff
@@ -21,24 +18,11 @@ func NewOptimizeSGCommand(args *inArgs) *cobra.Command {
 		Long:  `OptimizeSG attempts to reduce the number of security group rules in a SG without changing the semantic.`,
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, _ []string) error {
-			return optimization(cmd, args)
+			if err := validateFlags(args); err != nil {
+				return err
+			}
```

Comment:
If you run `validateFlags(args)` in each and every sub-command, maybe move this code to a `PersistentPreRun` in `rootCmd`.
(you'll have to add `cobra.EnableTraverseRunHooks = true` in `NewRootCommand()`)